org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/handlers/MoveHandlerTest.java

```diff
@@ -535,4 +536,70 @@ public void testMoveStaticInnerType() throws Exception {
 		assertNotNull(textEdit);
 		assertEquals(expected, TextEditUtil.apply(unitFoo.getSource(), textEdit.getEdits()));
 	}
+
+	@Test
+	public void testMoveInnerTypeToFile() throws Exception {
+		System.setProperty("line.separator", "\n");
+		IPackageFragment pack1 = sourceFolder.createPackageFragment("test1", false, null);
+		//@formatter:off
+		ICompilationUnit cu = pack1.createCompilationUnit("Top.java", "package test1;\n"
+				+ "\n"
+				+ "public class Top {\n"
+				+ "    String name;\n\n"
+				+ "    public class Inner {\n"
+				+ "        public void print() {\n"
+				+ "            System.out.println(Top.this.name);\n"
+				+ "        }\n"
+				+ "    }\n"
+				+ "}", false, null);
+		//@formatter:on
+
+		CodeActionParams params = CodeActionUtil.constructCodeActionParams(cu, "class Inner");
+		RefactorWorkspaceEdit refactorEdit = MoveHandler.move(new MoveParams("moveTypeToNewFile", params, "Foo", true), new NullProgressMonitor());
+		assertNotNull(refactorEdit);
+		assertNotNull(refactorEdit.edit);
+		List<Either<TextDocumentEdit, ResourceOperation>> changes = refactorEdit.edit.getDocumentChanges();
+		assertEquals(3, changes.size());
+
+		//@formatter:off
+		String expected = "package test1;\n"
+						+ "\n"
+						+ "public class Top {\n"
+						+ "    String name;\n"
+						+ "}";
+		//@formatter:on
+		TextDocumentEdit textEdit = changes.get(0).getLeft();
+		assertNotNull(textEdit);
+		assertEquals(expected, TextEditUtil.apply(cu.getSource(), textEdit.getEdits()));
+
+		ResourceOperation resourceOperation = changes.get(1).getRight();
+		assertNotNull(resourceOperation);
+		assertTrue(resourceOperation instanceof CreateFile);
+		assertEquals(ResourceUtils.fixURI(cu.getResource().getRawLocationURI()).replace("Top", "Inner"), ((CreateFile) resourceOperation).getUri());
+
+		//@formatter:off
+		expected = "package test1;\n"
+				+ "\n"
+				+ "public class Inner {\n"
+				+ "    /**\n"
+				+ "	 *\n"
+				+ "	 */\n"
+				+ "	\n"

```

Comment:
javadoc has an extra empty line. Doesn't happen in Eclipse

Reply:
Nice catch!
The cause is we added an extra ending empty line in CodeGenerationTemplate.java. Remove it will fix this issue.
```java
	/**
	 * Default value for field comments
	 */
	public static final String CODETEMPLATE_FIELDCOMMENT_DEFAULT = "/**\n" + " *\n" + " */\n";

	/**
	 * Default value for constructor comments
	 */
	public static final String CODETEMPLATE_CONSTRUCTORCOMMENT_DEFAULT = "/**\n" + " * ${tags}\n" + " */\n";

```